node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContext.java

```diff
@@ -27,6 +27,8 @@
 
     String vespaUser();
 
+    int vespaUserIdOnHost();
```

Comment:
So why is this ID for the host, but user for the container? We should use user everywhere since we already expect that in some other methods like `FileWriter` and `MakeDirectory`

Reply:
Doh! This is the user ID on host of `VESPA_USER` in the container, not user ID of `VESPA_USER` on host...

We can still require that any `VESPA_USER` in container maps to a user on the host, and this method could still return a username.

Maybe this should return some `ContainerVespaUser` object which has methods `userInContainer()` and `userOnHost()`/`userIdOnHost()`?

Reply:
If each container gets their own user ID range, I don't see a good way to define unique names for the user IDs, nor the need to do it if we can just operate on the ID?

Reply:
Yes, I just want some consistent APIs. We will likely use `FileWriter` for _something_ in `node-admin` soon, and it doesn't support UID.

That being said, I don't see a problem with defining unique names for each container. 
1. We don't need to define all of the names, maybe just `VESPA_USER`, maybe `root`?
2. Because its just 1/2 users, we can manually create it with `useradd` and the name would be `<container-name>_vespa`/`<container-name>_root`.
If the ranges are static for each container, we will create those users just once...

Reply:
Moved to using username.